pkg/internal/common/logging/modlog/provider.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package modlog
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	logapi "github.com/hyperledger/aries-framework-go/pkg/common/log"
+)
+
+//providerOpts contains options for initializing Moduled logging provider
+type providerOpts struct {
+	customProvider logapi.LoggerProvider
+}
+
+//ModuledLoggerOpts is option to provide logger provider opts for initializing Moduled logging provider
+type ModuledLoggerOpts func(opts *providerOpts)
+
+//WithCustomProvider can be used to provide custom logger provider for moduled logger
+func WithCustomProvider(customProvider logapi.LoggerProvider) ModuledLoggerOpts {
+	return func(opts *providerOpts) {
+		opts.customProvider = customProvider
+	}
+}
+
+//ModuledLoggerProvider returns logger provider for moduled level based logger
+func ModuledLoggerProvider(opts ...ModuledLoggerOpts) logapi.LoggerProvider {
+	providerOpts := &providerOpts{}
+	for _, opt := range opts {
+		opt(providerOpts)
+	}
+	return &provider{providerOpts.customProvider}
+}
+
+// provider is the default logger implementation
+type provider struct {
+	customProvider logapi.LoggerProvider
+}
+
+//GetLogger returns moduled logger implementation.
+// default logging built on top go log will be used if custom logger provider is not provided.
+// if custom logging provider is provided then moduled logger wrapper of custom logger will be returned.
+func (p *provider) GetLogger(module string) logapi.Logger {
+	if p.customProvider != nil {
```

Comment:
instead of an if condition, i would always set p.provider. 

i.e., in the case of defLog, you should have a defProvider.

Reply:
spilt it into --> NewModLogProvider & NewDefLogProvider